File: tests/manual_auth_browser.py
from playwright.sync_api import sync_playwright


BASE_URL = 'http://127.0.0.1:8788'
SECRET = 'local-auth-bootstrap-secret'
EMAIL = 'admin@example.com'
PASSWORD = 'correct horse battery staple'


with sync_playwright() as playwright:
    browser = playwright.chromium.launch(headless=True)
    page = browser.new_page()
    page.goto(BASE_URL)
    page.wait_for_load_state('networkidle')
    page.wait_for_selector('#awq-login-gate')

    bootstrap = page.request.post(BASE_URL + '/api/rpc', data={
        'method': 'authBootstrap',
        'args': [SECRET, EMAIL, PASSWORD],
    }, headers={'Origin': BASE_URL})
    assert bootstrap.status == 200, bootstrap.text()

    page.get_by_label('Email').fill(EMAIL)
    page.get_by_label('Password').fill(PASSWORD)
    page.get_by_role('button', name='Sign in').click()
    page.wait_for_selector('#awq-user-bar')
    assert page.locator('#awq-user-bar').inner_text().startswith('ADMIN')

    page.screenshot(path='test-results/auth-admin-logged-in.png', full_page=True)
    page.get_by_role('button', name='Logout').click()
    page.wait_for_selector('#awq-login-gate')
    page.screenshot(path='test-results/auth-logged-out.png', full_page=True)
    browser.close()

print('Browser auth smoke test passed: bootstrap, login, account bar, logout, and login gate.')

File: tests/manual_auth_browser.mjs
import { chromium } from 'playwright';

const baseUrl = 'http://127.0.0.1:8788';
const browser = await chromium.launch({ headless: true });
const page = await browser.newPage();
page.on('console', message => console.log('[browser]', message.type(), message.text()));
console.log('opening page');
const bootstrap = await page.request.post(baseUrl + '/api/rpc', {
  data: { method: 'authBootstrap', args: ['local-auth-bootstrap-secret', 'admin@example.com', 'correct horse battery staple'] },
  headers: { Origin: baseUrl }
});
if (bootstrap.status() !== 200) throw new Error('Bootstrap failed: ' + await bootstrap.text());
console.log('bootstrap passed');
await page.goto(baseUrl + '/app');
await page.waitForSelector('#awq-login-gate', { timeout: 15000 });
console.log('login gate visible');
await page.locator('#awq-login-form').evaluate(form => {
  form.querySelector('#awq-login-email').value = 'admin@example.com';
  form.querySelector('#awq-login-password').value = 'correct horse battery staple';
  form.dispatchEvent(new Event('submit', { bubbles: true, cancelable: true }));
});
console.log('login submitted');
await page.waitForSelector('#awq-user-bar');
console.log('login passed');
if (!(await page.locator('#awq-user-bar').innerText()).startsWith('ADMIN')) throw new Error('Admin bar missing');
await page.locator('#awq-logout').dispatchEvent('click');
await page.waitForSelector('#awq-login-gate');
await browser.close();
console.log('Browser auth smoke test passed.');
